test/test_autograd.py

```diff
@@ -10804,11 +10804,42 @@ def hook(*_unused_args):
             x.backward(retain_graph=True)
 
         a = torch.tensor(1., requires_grad=True)
-        with torch.utils.checkpoint._set_checkpoint_early_stop(early_stop):
+        with torch.utils.checkpoint.set_checkpoint_early_stop(early_stop):
             x, out = checkpoint(fn, a, use_reentrant=False)
         out.grad_fn.register_hook(hook)
         out.backward(retain_graph=True)
 
+    def test_nested_checkpoint_set_early_stop(self):
+        counter = [0]
+
+        def clone(x):
+            counter[0] += 1
+            return x.clone()
+
+        def fn(x):
+            # Since clone does not save anything, it is not recomputed iff
```

Comment:
btw do we have tests to make sure the case where nothing needs to be recomputed works fine?
With non-reentrant it should be trivially working but still good to have a test.